### PR TITLE
fix: Update support for `Layers`

### PR DESCRIPTION
**Pull request by Vandita2020**
**Description:**
This PR adds the missing update feature for `Layers` property in Function resource.

**Acknowledgement:**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/function/hooks.go [hunk 1]
```diff
@@ -179,6 +179,38 @@ func (rm *resourceManager) updateFunctionConfiguration(
...
+				elemCopy := elem.DeepCopy()
+				fscElem := &svcsdk.FileSystemConfig{
+					Arn:            elemCopy.ARN,
+					LocalMountPath: elemCopy.LocalMountPath,
+				}
+				fileSystemConfigs = append(fileSystemConfigs, fscElem)
+			}
+			input.FileSystemConfigs = fileSystemConfigs
+		}
+	}
```
Comment (a-hilaly) [inline pkg/resource/function/hooks.go:212]: Thanks @Vandita2020 ! How are environement/ephemeralStorage/FileSystemConfigs/Timeouts affecting to layers?

File: templates/hooks/function/sdk_read_one_post_set_output.go.tpl [hunk 2]
```diff
@@ -29,7 +24,9 @@
 				f16elem.SigningProfileVersionARN = f16iter.SigningProfileVersionArn
 			}
 			f16 = append(f16, f16elem)
+			layer = append(layer, f16iter.Arn)
 		}
+		ko.Spec.Layers = layer
```
Comment (a-hilaly) [inline templates/hooks/function/sdk_read_one_post_set_output.go.tpl:29]: Can we have some smoke tests around layers updates?

File: pkg/resource/function/hooks.go (hunk 1 above)
Comment (Vandita2020) [reply pkg/resource/function/hooks.go:212]: Hey, I have just rearranged them according to the order in the API. Please ignore them, no changes for them.

File: test/e2e/tests/test_function.py [hunk 3]
```diff
@@ -664,6 +664,69 @@ def test_function_features(self, lambda_client):
...
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        lambda_validator = LambdaValidator(lambda_client)
+
+        # Check Lambda function exists
+        assert lambda_validator.function_exists(resource_name)
+
+        # Update cr
+        layers_list = ["arn:aws:lambda:us-west-2:017000801446:layer:AWSLambdaPowertoolsPythonV2:68", "arn:aws:lambda:us-west-2:580247275435:layer:LambdaInsightsExtension:52"]
```
Comment (a-hilaly) [inline test/e2e/tests/test_function.py:710]: qq: I guess those are official layers that anybody can reference?

**Review by a-hilaly — APPROVED** [review]
Thank you @Vandita2020 !
/lgtm

File: test/e2e/tests/test_function.py (hunk 3 above)
Comment (Vandita2020) [reply test/e2e/tests/test_function.py:710]: Yes, those are from the list of layers provided by AWS.

**Pull request merged**